api/envoy/config/filter/network/redis_proxy/v2/redis_proxy.proto

```diff
@@ -48,10 +54,58 @@ message RedisProxy {
     bool enable_hashtagging = 2;
   }
 
-  // Network settings for the connection pool to the upstream cluster.
+  // Network settings for the connection pool to the upstream clusters.
   ConnPoolSettings settings = 3 [(validate.rules).message.required = true];
 
   // Indicates that latency stat should be computed in microseconds. By default it is computed in
   // milliseconds.
   bool latency_in_micros = 4;
+
+  message PrefixRoutes {
+    message Route {
+      // String prefix that must match the beginning of the keys. Envoy will always favor the
+      // longest matches.
+      string prefix = 1 [(validate.rules).string.min_bytes = 1];
+
+      // Indicates if the prefix needs to be removed from the key when forwarded.
+      bool remove_prefix = 2;
+
+      // Upstream cluster to forward the command to.
+      string cluster = 3;
+    }
+
+    // List of prefix routes.
+    repeated Route routes = 1 [(gogoproto.nullable) = false];
+
+    // Indicates that prefix matching should be case insensitive.
+    bool case_insensitive = 2;
+
+    // Optional catch-all route to forward commands that doesn't match any of the routes. The
+    // catch-all route becomes required when no routes are specified.
+    string catch_all_cluster = 3;
+  }
+
+  // List of prefixes used to separate keys from different workloads to different clusters. Envoy
+  // will always favor the longest matches first. Prefixes are stored in a Trie and are accessed
+  // using a LIFO, thus identical prefixes will always resolve to the last entry inserted. A
+  // catch-all cluster can be used to forward commands to when there is no match. Time complexity of
+  // the lookups are in O(min(longest key prefix, key length)).
+  //
+  // Example:
+  //
+  // .. code-block:: yaml
+  //
+  //    prefix_routes:
+  //      routes:
+  //        - prefix: "abc" # this prefix will be overwritten by the next entry
```

Comment:
Yeah I would go ahead and add some way to detect this duplication and fail. I think it would be a better user experience and make this example unecessary and the description above simpler to understand. WDYT?